.github/workflows/build-wheels.yml

```diff
@@ -0,0 +1,197 @@
+    needs: [determine-source-date-epoch]
+    name: "Wheel awkward-cpp: ${{ matrix.type }} ${{ matrix.arch }} on ${{ matrix.os }} with ${{ matrix.build }}"
+    runs-on: ${{ matrix.os }}
+    env:
+      SOURCE_DATE_EPOCH: ${{ needs.determine-source-date-epoch.outputs.source-date-epoch }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+        arch: [auto64]
+        build: ["cp", "pp"]
+
+        include:
+        - os: macos-latest
+          type: "Universal"
+          arch: universal2
+          build: "cp"
+
+        - os: windows-latest
+          arch: auto64
+          build: "cp"
+
+        - os: windows-latest
+          arch: auto32
+          build: "cp{38,39}-"
+
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        submodules: true
+
+    - name: Python 3.11
+      uses: actions/setup-python@v5
+      with:
+        python-version: '3.11'
+
+    - name: Prepare build files
+      run: pipx run nox -s prepare
+
+    - uses: pypa/cibuildwheel@v2.16.5
+      env:
+        CIBW_BUILD: "${{ matrix.build }}*"
+        CIBW_ARCHS: ${{ matrix.arch }}
+      with:
+        config-file: cibuildwheel.toml
+        package-dir: awkward-cpp
+
+    - name: Check metadata
+      shell: python
+      run: |
+        import subprocess, glob
+        subprocess.run(
+          ["pipx", "run", "twine", "check", *glob.glob("wheelhouse/*.whl")],
+          check=True
+        )
+
+    - name: Upload wheels
+      uses: actions/upload-artifact@v4
+      with:
+        name: awkward-cpp-wheels-${{ matrix.os }}-${{ matrix.arch }}-${{ matrix.build }}
+        path: wheelhouse/*.whl
+
+  build_alt_wheels:
+    needs: [determine-source-date-epoch]
+    name: "Wheel awkward-cpp: ${{ matrix.python }} on ${{ matrix.arch }}"
+    runs-on: ubuntu-latest
+    env:
+      SOURCE_DATE_EPOCH: ${{ needs.determine-source-date-epoch.outputs.source-date-epoch }}
+    strategy:
+      matrix:
+        python: [38, 39, 310, 311, 312]
+        arch: [aarch64]
+    steps:
+
+    - uses: actions/checkout@v4
+      with:
+        submodules: true
+
+    - name: Python 3.10
+      uses: actions/setup-python@v5
+      with:
+        python-version: '3.10'
+
+    - name: Prepare build files
+      run: pipx run nox -s prepare
+
+    - uses: docker/setup-qemu-action@v3.0.0
+
+    - uses: pypa/cibuildwheel@v2.16.5
+      env:
+        CIBW_BUILD: cp${{ matrix.python }}-*
+        CIBW_ARCHS: ${{ matrix.arch }}
+      with:
+        config-file: cibuildwheel.toml
+        package-dir: awkward-cpp
+
+    - name: Check metadata
+      shell: python
+      run: |
+        import subprocess, glob
+        subprocess.run(
+          ["pipx", "run", "twine", "check", *glob.glob("wheelhouse/*.whl")],
+          check=True
+        )
+
+    - name: Upload wheels
+      uses: actions/upload-artifact@v4
+      with:
+        name: awkward-cpp-wheels-${{ matrix.arch }}-py${{ matrix.python }}
+        path: wheelhouse/*.whl
+
+  build_awkward_wheel:
```

Comment:
This only ever gets used for the nightly builds. The pure-Python sdist and wheel get rebuilt in the `deploy.yml` workflow as it is so quick.